feat(dataset-compare): manually review in compare view

marliessophie opened the pull request:
<!-- ELLIPSIS_HIDDEN -->


> [!IMPORTANT]
> Introduces manual review in dataset compare view with annotation panel, score caching, and enhanced annotation form.
> 
>   - **Behavior**:
>     - Adds manual review feature in dataset compare view with `AnnotationPanel` and `SidePanel` in `index.tsx`.
>     - Introduces `ActiveCellContext` to track active dataset run item for annotation.
>     - Implements score caching with `ScoreCacheContext` for optimistic UI updates.
>     - Enhances `AnnotationForm` to support both fixed and selectable score configs.
>   - **Caching and Merging**:
>     - Adds `useMergedScores`, `useMergedAggregates`, and `useMergedAnnotationScores` hooks for merging server data with cache.
>     - Implements `mergeScoresWithCache`, `mergeAggregatesWithCache`, and `mergeAnnotationScoresWithCache` for cache operations.
>   - **Components and Hooks**:
>     - Refactors `AnnotateDrawer` to use `AnnotationForm`.
>     - Updates `ScoreRow` to display score details with hover effects.
>     - Adds `useScoreMutations` for handling score create, update, and delete with cache updates.
>     - Introduces `useScoreConfigSelection` and `useAnnotationScoreConfigs` for managing score config selection.
>   - **Misc**:
>     - Updates `scores.ts` API router to handle new score operations.
>     - Refactors score-related types and schemas for better integration.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=langfuse%2Flangfuse&utm_source=github&utm_medium=referral)<sup> for 6f45706b1b2a8af4e0061a7200d1d253198aaa8b. You can [customize](https://app.ellipsis.dev/langfuse/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

marliessophie committed:
chore: remove peek view from compare data table

marliessophie committed:
style(io-table-cell): support content expand on hover

marliessophie committed:
feat: make compare cell digestable

marliessophie committed:
feat: open trace peek view from compare view

marliessophie committed:
style: adjust cell for many scores, IO and metadata

marliessophie committed:
style: allow untoggling output

marliessophie committed:
chore: fix

marliessophie committed:
Merge branch 'main' into marlies/lfe-6975-compare-view-cell

marliessophie committed:
fix: enable score column query based on runIds length

marliessophie committed:
feat: add ScoreWriteCache context for optimistic UI updates on score edits

marliessophie committed:
feat: add ActiveCellContext for managing active dataset run item cell state

marliessophie committed:
feat: enhance SidePanel component with controlled/uncontrolled modes

marliessophie committed:
feat: add annotation functionality to DatasetAggregateCell component

marliessophie committed:
feat: integrate annotation panel and context providers in DatasetCompare component

marliessophie committed:
chore: add access control for annotation button in DatasetAggregateCell component

marliessophie committed:
fixup: add annotation panel as sidepanel child

marliessophie committed:
chore: extract useConfigSelection

marliessophie committed:
chore: refactor AnnotateDrawerContent to use new useAnnotationFormHandlers for score management

marliessophie committed:
refactor: simplify score mutation handlers by consolidating query invalidation logic

marliessophie committed:
Merge branch 'main' into marlies/lfe-4577

marliessophie committed:
chore: refactor to simplify AnnotateDrawer props

marliessophie committed:
chore: extract useEmptyConfigs hook

marliessophie committed:
chore: transform score aggregates to annotation scores

marliessophie committed:
feat: enhance annotation score handling with new caching and filtering mechanisms

marliessophie committed:
test: transform and filter scores

marliessophie committed:
chore: delete

marliessophie committed:
style: paddings and margins

marliessophie committed:
fix: ensure cached updates apply to cached creates

marliessophie committed:
chore: do not allow closing annotation panel if has draft comment

marliessophie committed:
chore: refactor mergeScoreAggregateWithCache for readability

marliessophie commented on web/src/features/scores/hooks/useConfigSelection.ts:
Extracted to custom hook, no need to review, no changes.

marliessophie commented on web/src/features/scores/lib/annotationFormHelpers.ts:
Extracted to helper file only, no need to review.

marliessophie committed:
fix: update dependency in useEffect to use router.query for accurate data handling

marliessophie committed:
refactor: optimize DatasetAggregateCell for performance using MemoizedIOTableCell and useMemo

marliessophie committed:
Merge branch 'main' into marlies/lfe-4577

marliessophie committed:
fix: test

marliessophie committed:
chore: fix build

marliessophie committed:
chore: move files

marliessophie committed:
chore: fix pending creates logic

marliessophie committed:
refactor: enhance mergeScoreAggregateWithCache to prioritize cached creates and streamline aggregate handling

marliessophie committed:
fix: add comment field to annotation form handlers

marliessophie committed:
refactor: update AnnotationPanel and related components to utilize filterSingleValueAggregates for improved score handling and disabled config management

marliessophie committed:
refactor: optimize score field rendering in AnnotateDrawerContent by sorting scores alphabetically

marliessophie committed:
refactor: merge score columns in DatasetAggregateCell

marliessophie committed:
test: fix

marliessophie committed:
style: unify compare cell output heights

maxdeichmann commented on web/src/components/trace/TracePreview.tsx:
```diff
@@ -73,9 +74,8 @@ export const TracePreview = ({
-  const [emptySelectedConfigIds, setEmptySelectedConfigIds] = useLocalStorage<
-    string[]
-  >("emptySelectedConfigIds", []);
+  const { emptySelectedConfigIds, setEmptySelectedConfigIds } =
```
Doesnt it make sense to move this inside the drawer? Is there any case where this is not required?

maxdeichmann commented:
<img width="984" height="357" alt="image" src="https://github.com/user-attachments/assets/caae84d4-4fd8-47a5-9cdf-4c75f5015784" />

Great idea to select the DRI to be annotated. I think we did not use the blue anywhere yet. Could you please take a color that matches better?

marliessophie committed:
Merge branch 'main' into marlies/lfe-4577

marliessophie committed:
refactor(scores): replace useEmptyConfigs with scoreMetadata in annotation components

- Removed useEmptyConfigs hook and its related state from multiple components including SessionPage, ObservationPreview, and TracePreview.
- Introduced scoreMetadata prop to pass projectId and environment directly to annotation components.
- Updated AnnotateDrawer and AnnotationForm to accommodate the new scoreMetadata structure.
- Refactored related components to streamline score handling and improve performance by eliminating unnecessary state management.

marliessophie committed:
refactor(scores): update score schemas and cache handling

- Made `id` and `configId` fields mandatory in `CreateAnnotationScoreBase` and `AnnotationScoreDataSchema` for consistency.
- Removed `useEmptyConfigs` imports from components to streamline code.
- Enhanced `ScoreCacheContext` to manage cache operations more effectively, including adding, updating, and deleting scores.
- Updated hooks and components to utilize the new cache methods, improving performance and reducing unnecessary state management.
- Added TODO comments for future schema reviews and adjustments.

marliessophie committed:
chore: remove unused type

marliessophie committed:
refactor(scores): enhance score handling and component structure

- Made `id` optional in `CreateAnnotationScoreBase` for backward compatibility.
- Updated `Trace` component to utilize `useMergedScores` for improved score management.
- Refactored `SpanItem` to use loose equality check for `observationId`.
- Changed `AnnotationDrawerSection` to accept `configSelection` prop for better config handling.
- Simplified `AnnotationPanel` by removing unnecessary API calls and directly using active cell data.
- Renamed `useEmptyConfigs` to `useEmptyScoreConfigs` for clarity and updated related components.
- Introduced `useAnnotationScoreConfigs` hook to manage score config selection logic.
- Removed unused `useScoreCustomOptimistic` and `useScoreValues` hooks to clean up the codebase.

marliessophie committed:
chore: types

marliessophie committed:
chore: types

marliessophie committed:
chore: types

marliessophie committed:
chore: types

marliessophie committed:
refactor(scores): integrate score column merging and caching

marliessophie committed:
fix(scores): add error handling to score mutations

- Introduced an `onError` callback to reload the page upon mutation errors, ensuring cache invalidation and fresh data retrieval.
- Updated import for `ScoreTarget` type from `@langfuse/shared` for consistency.

marliessophie committed:
chore: types

marliessophie committed:
chore: change color of highlight cell

marliessophie committed:
style: margins

marliessophie committed:
fixup: remove alphabetic sorting

maxdeichmann commented on web/src/components/trace/SpanItem.tsx:
```diff
@@ -157,14 +157,14 @@ export const SpanItem: React.FC<SpanItemProps> = ({
               scores={
                 node.type === "TRACE"
-                  ? scores.filter((s) => s.observationId === null)
+                  ? scores.filter((s) => s.observationId == null)
```
You switched to loose equality here. I think we should avoid this.

maxdeichmann commented on web/src/pages/project/[projectId]/traces/[traceId].tsx:
```diff
@@ -10,5 +11,9 @@ export default function Trace() {
+  return (
+    <ScoreCacheProvider>
+      <TracePage traceId={traceId} timestamp={timestamp} />
+    </ScoreCacheProvider>
```
Is this still required as the context is already set in _app.tsx?

marliessophie committed:
fix: sorting

marliessophie committed:
chore: drop context provider from traces component

marliessophie committed:
chore: updated `useMergedScores` and related functions to accept a `mode` parameter for better control over score merging.

- Adjusted components to utilize the new `displayScores` for improved performance and user experience.

marliessophie committed:
chore: resolve cached score from score domain

marliessophie committed:
chore: lint

marliessophie committed:
chore: fix build

marliessophie committed:
fix: null vs undefined comparison

marliessophie committed:
Merge branch 'main' into marlies/lfe-4577
